source/adios2/toolkit/interop/hdf5/HDF5Common.cpp

```diff
@@ -64,6 +64,12 @@ HDF5Common::HDF5Common(const bool debugMode) : m_DebugMode(debugMode)
     H5Tinsert(m_DefH5TypeComplexDouble, "dimg", H5Tget_size(H5T_NATIVE_DOUBLE),
               H5T_NATIVE_DOUBLE);
 
+    m_DefH5TypeComplexLongDouble =
+        H5Tcreate(H5T_COMPOUND, sizeof(std::complex<long double>));
+    H5Tinsert(m_DefH5TypeComplexLongDouble, "dreal", 0, H5T_NATIVE_LDOUBLE);
```

Comment:
note: I would generally vote to rename those labels to `r` and `i` as it is convention by `h5py`
cc @guj

Reply:
@guj Looks to me like we forget to free those types again in the end, via `H5Tclose`.

Reply:
I did these in pull request 1913.  not merged yet as I see you made changes in the same  files.  might have  conflict.

Reply:
Thanks, added your commits in #1913 to this PR :)